examples/contract-terminate/lib.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright 2018-2020 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A smart contract which demonstrates behavior of the `self.env().terminate()`
+//! function. It terminates itself once `terminate_me()` is called.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract]
+pub mod just_terminates {
+    /// No storage is needed for this simple contract.
+    #[ink(storage)]
+    pub struct JustTerminate {}
+
+    impl JustTerminate {
+        /// Creates a new instance of this contract.
+        #[ink(constructor)]
+        pub fn new() -> Self {
+            Self {}
+        }
+
+        /// Terminates with the caller as beneficiary.
+        #[ink(message)]
+        pub fn terminate_me(&mut self) {
+            self.env().terminate_contract(self.env().caller());
+        }
```

Comment:
The correct version of this would be:
```suggestion
        pub fn terminate_me(&mut self) -> ! {
            self.env().terminate_contract(self.env().caller())
        }
```

We then run into `error[E0277]: the trait bound `!: scale_info::TypeInfo` is not satisfied` though. Wdyt?